src/app/main.cpp

```diff
@@ -793,32 +815,65 @@ int main( int argc, char *argv[] )
     exit( 1 ); //exit for now until a version of qgis is capabable of running non interactive
   }
 
-  if ( !optionpath.isEmpty() || !configpath.isEmpty() )
-  {
-    // tell QgsSettings to use INI format and save the file in custom config path
-    QSettings::setDefaultFormat( QSettings::IniFormat );
-    QSettings::setPath( QSettings::IniFormat, QSettings::UserScope, optionpath.isEmpty() ? configpath : optionpath );
-  }
-
   // GUI customization is enabled according to settings (loaded when instance is created)
   // we force disabled here if --nocustomization argument is used
   if ( !myCustomization )
   {
     QgsCustomization::instance()->setEnabled( false );
   }
 
-  QgsApplication myApp( argc, argv, myUseGuiFlag, configpath );
-
-  myApp.setWindowIcon( QIcon( QgsApplication::appIconPath() ) );
-
-
-  //
+  QgsApplication myApp( argc, argv, myUseGuiFlag );
   // Set up the QSettings environment must be done after qapp is created
   QCoreApplication::setOrganizationName( QgsApplication::QGIS_ORGANIZATION_NAME );
   QCoreApplication::setOrganizationDomain( QgsApplication::QGIS_ORGANIZATION_DOMAIN );
   QCoreApplication::setApplicationName( QgsApplication::QGIS_APPLICATION_NAME );
   QCoreApplication::setAttribute( Qt::AA_DontShowIconsInMenus, false );
 
+  QgsUserConfig *config;
+
+  // As of QGIS 3 all QGIS settings will come from QGIS\config
+  // default config will be loaded from .default file
+  // which contains the name of the startup default config.
+
+  if ( !configLocalStorageLocation.isEmpty() )
```

Comment:
@m-kuhn adding a override `--config-storage-location path` to allow for setting a custom location.